src/cli/commands/install.js

```diff
@@ -494,7 +494,9 @@ export class Install {
     }
 
     // fin!
-    await this.saveLockfileAndIntegrity(topLevelPatterns, workspaceLayout);
+    if (topLevelPatterns.length || await fs.exists(path.join(this.config.cwd, constants.LOCKFILE_FILENAME))) {
+      await this.saveLockfileAndIntegrity(topLevelPatterns, workspaceLayout);
+    }
```

Comment:
I think I missed this one, do you need to check `await fs.exists(path.join(this.config.cwd, constants.LOCKFILE_FILENAME)`?